src/program.ts

```diff
@@ -3291,12 +3293,35 @@ export class Parameter {
   ) {}
 }
 
+export class ClosedLocal extends VariableLikeElement {
+  local: Local;
+  functionDepth: i32 = 1;
+
+  constructor(local: Local) {
+    super(
+      ElementKind.CLOSED_LOCAL,
+      local.name,
+      local.parent,
+      <VariableLikeDeclarationStatement>local.declaration
+    )
+    this.local = local;
+  }
+
+  lookup(name: string) {
+    return this.local.lookup(name);
+  }
+}
```

Comment:
This is meant to act as a wrapper over a Local to add the additional contextual information of the `functionDepth` which tells us how many environments up we'll need to go before we can find this local

Reply:
Introducing an on-demand wrapper like this makes sense to me, ideally also deduplicating these wrappers per function as mentioned further down.